web/frontend/src/pages/ballot/QuestionDisplay.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Answers,
+  Question,
+  RankAnswer,
+  RankQuestion,
+  SelectAnswer,
+  SelectQuestion,
+  Subject,
+  TextQuestion,
+} from 'types/configuration';
+import { RankDisplay } from './components/RankDisplay';
+import { Droppable } from 'react-beautiful-dnd';
+import { TextDisplay, TextHintDisplay } from './components/TextDisplay';
+import { SelectDisplay, SelectHintDisplay } from './components/SelectDisplay';
+
+export type HintDisplayProps = {
+  questionContent: RankQuestion | TextQuestion | SelectQuestion | Subject;
+};
+
+export function renderRank(
+  question: Question,
+  answers: Answers,
+  setAnswers: React.Dispatch<React.SetStateAction<Answers>>
+) {
+  let rank = question.Content as RankQuestion;
+
+  return (
+    <div>
+      <h3 className="text-lg text-gray-600">{rank.Title}</h3>
+      <div className="mt-5 pl-8">
+        <Droppable droppableId={String(rank.ID)}>
+          {(provided) => (
+            <ul
+              className={question.Content.ID}
+              {...provided.droppableProps}
+              ref={provided.innerRef}>
+              {Array.from(
+                answers.RankAnswers.find((r: RankAnswer) => r.ID === rank.ID).Answers.entries()
+              ).map(([rankIndex, choiceIndex]) => (
+                <RankDisplay
```

Comment:
Be careful when using lists and maps: https://reactjs.org/docs/lists-and-keys.html#keys